OpenSim/Simulation/Test/testInverseKinematicsSolver.cpp

```diff
@@ -0,0 +1,457 @@
+void testUpdateMarkerWeights();
+// Verify that the track() solution is also effected by updating marker
+// weights and marker error is being reduced as its weighting increases.
+void testTrackWithUpdateMarkerWeights();
+
+int main()
+{
+    SimTK::Array_<std::string> failures;
+
+    try { testAccuracy(); }
+    catch (const std::exception& e) {
+        cout << e.what() << endl; failures.push_back("testAccuracy");
+    }
+    try { testUpdateMarkerWeights(); }
+    catch (const std::exception& e) {
+        cout << e.what() << endl; failures.push_back("testUpdateGoalWeights");
+    }
+    try { testTrackWithUpdateMarkerWeights(); }
+    catch (const std::exception& e) {
+        cout << e.what() << endl; failures.push_back("testAccuracy");
+    }
+
+    if (!failures.empty()) {
+        cout << "Done, with failure(s): " << failures << endl;
+        return 1;
+    }
+
+    cout << "Done. All cases passed." << endl;
+
+    return 0;
+}
+
+//=============================================================================
+// Test Cases
+//=============================================================================
+
+void testAccuracy()
+{
+    cout << "\ntestInverseKinematicsSolver::testAccuracy()" << endl;
+
+    std::unique_ptr<Model> pendulum{ constructPendulumWithMarkers() };
+    Coordinate& coord = pendulum->getCoordinateSet()[0];
+
+    double refVal = 0.123456789;
+    double looseAccuracy = 1e-3;
+    double tightAccuracy = 1.0e-9;
+
+    SimTK::Array_<CoordinateReference> coordRefs;
+    Constant coordRefFunc(refVal);
+    CoordinateReference coordRef(coord.getName(), coordRefFunc);
+    coordRef.setWeight(1.0);
+    coordRefs.push_back(coordRef);
+
+    SimTK::State state = pendulum->initSystem();
+    double coordValue = coord.getValue(state);
+    double refVal0 = coordRefs[0].getValue(state);
+
+    cout.precision(10);
+    cout << "Initial " << coord.getName() << " value = " << coordValue <<
+        " referenceValue = " << refVal0 << endl;
+
+    coord.setValue(state, refVal);
+    StatesTrajectory states;
+    states.append(state);
+
+    MarkerData* markerData =
+        generateMarkerDataFromModelAndStates(*pendulum, states);
+    // MarkersReference takes ownership of the markerData
+    MarkersReference markersRef(markerData);
+    markersRef.setDefaultWeight(1.0);
+
+    // Reset the initial coordinate value
+    coord.setValue(state, 0.0);
+    InverseKinematicsSolver ikSolver(*pendulum, markersRef, coordRefs);
+    ikSolver.setAccuracy(looseAccuracy);
+    ikSolver.assemble(state);
+
+    coordValue = coord.getValue(state);
+    cout << "Assembled " << coord.getName() << " value = " << coordValue << endl;
+
+    double accuracy = abs(coordValue - refVal);
+    cout << "Specified accuracy: " << looseAccuracy << "; achieved: "
+        << accuracy << endl;
+
+    // test should fail if we did not meet the target accuracy
+    SimTK_ASSERT_ALWAYS(accuracy <= looseAccuracy,
```

Comment:
Sorry, I misinterpreted this as if it were `OPENSIM_THROW_IF()`, I think because the comment is written that way around (i.e., conditions for failure). No changes necessary.